content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+SVG can be inlined by turning an `.svg` file into a [React][1] Component (An SVG file is just a regular HTML tree)
+or by using Webpack's [inline-svg-loader][101]
+
+The main advantage of inlining is that after you bundle all your code base you end up with
+a single Javascript file that contains all the dependencies it needs to run, even CSS and Images.
+
+This makes _very_ easy to package and publish to [NPM][102] and it makes very easy
+for **Component Consumers** to _use_ your module.
+
+Using **CSS-in-JS** libraries like [Emotion][5] or [Styled-components][6] makes up for a very
+nice **Run-Time** **Theming** API. Since everything is Javascript your Themes are regular Javascript
+object that can be composed or broken down into parts as any other regular JS object,
+and these can be altered at Run-Time, Themes can be composed, and they provide
+a much simpler strategy than Build-Time CSS generated Themes like [Bootstrap][9] or [Ant Design][8].
+[Material UI React][7] uses a Run-Time Theme via a custom **CSS-in-JS** solution.
+
+Inlining is particularly useful for Component libraries or even UI Frameworks because
+they tend to be mostly CSS heavy and with a couple of SVGs at most, both of which are efficiently
+inlined.
+
+Images are a bit more problematic because they are inlined in [base 64][103] making them 30% bigger.
+
+In my experience, most Component Libraries do not tend to use an over abundance
+of images and if they do they tend to be rather small.
+
+Most high resolution images are dynamic content that are usually dealt by
+Applications and not libraries, although libraries can render images, they do not
+ship with those images since they are _Content_ provided by APIs
+or static files in the App.
+
+The main drawback of inlining is that you turn multiple files that can
+be fetched in parallel by the browser to a single larger file that cannot,
+this will become more noticeable in larger code bases such as Big Component Libraries
+or Applications.
+
+If you have a big enough library of components you might able to break it
+into independent chunks that can be dynamically imported by Applications,
+allowing more parallelism.
+This is actually not a super far fetched solution since dynamic imports
+are becoming [widely available][104] and Webpack has been making good
+progress towards an easier configuration for multiple chunks.
+
+### Not-Inline
+
+#### As Component Authors
+
+The configuration is pretty standard and you can use
+already made solutions such as [create-react-library][105].
+
+You will end up with a bundle made of different files.
+
+A typical output might look like this:
+
+```
+/dist
+  - bundle.js
+  - bundle.css
+  images/
+    - dog.jpg
+```
+
+This is what you as an Author would publish to [NPM][102].
+
+`bundle.css` will use the image by doing `url(./images/dog.jpg)`, so
+both files need to be served in the following URLs
```

Comment:
nit: Pretty sure I've read "served on [a URL]" and "served from [a URL]" but never "served in"